content/news/posts/2025/breaking-change-guzzle-removal-v9.md

```diff
@@ -0,0 +1,77 @@
+---
+layout: post
+title:  "Adapting to PrestaShop 9: Guzzle is gone, what's next?"
+subtitle: "PrestaShop 9 no longer includes Guzzle by default. Here's how to adapt your module."
+date:   2025-03-18
+authors: [ JonathanLelievre ]
+icon: icon-code
+tags: [autoupgrade, module, beta, releases, community, contribution, development, qa]
+---
+
+PrestaShop 9 is a major release, meaning it [introduces breaking changes](https://devdocs.prestashop-project.org/9/modules/core-updates/9.0/), including the removal of certain dependencies. One of the most impactful changes for module developers is the removal of Guzzle, a popular HTTP client that many modules relied on.
+
+If your module depends on Guzzle, it will no longer be available by default in PrestaShop 9. In this article, we’ll explain why Guzzle was removed, what this means for your module, and how to migrate to a suitable alternative.
+
+## Why remove Guzzle in the first place?
+
+For many years, PrestaShop included Guzzle as a core dependency, allowing modules to use it for external HTTP requests. However, the core and PrestaShop tools used in it have gradually moved to using Symfony's HTTP Client instead, leaving Guzzle largely unused.
+
+Since PrestaShop no longer relies on Guzzle, keeping it in the core is unnecessary. Additionally, dependency management has been a recurring issue. In PrestaShop 8, upgrading Guzzle from version 5 to 7 already caused compatibility problems for modules. Some developers had to bundle their own Guzzle version, but when multiple modules included different versions, it led to conflicts. To prevent such issues and improve overall stability, we decided to remove Guzzle entirely.
+
+## How can I adapt my module?
+
+With Guzzle no longer available by default, you must adjust your module. Here are a few solutions to replace it effectively.
+
+### Use Symfony HTTP Client (recommended)
+
+The best way to avoid conflicts is to minimize external dependencies and rely on core libraries whenever possible. Since Symfony is the foundation of PrestaShop and will continue to be so in the future, it's a safe and stable dependency that won't be removed.
```

Comment:
```suggestion
The best way to avoid conflicts is to minimize external dependencies and rely on core libraries whenever possible. Since Symfony is the foundation of PrestaShop and will continue to be in the future, it's a safe and stable dependency that won't be removed.
```